app/models/collection.rb

```diff
@@ -19,8 +19,8 @@ class Collection < ActiveRecord::Base
   ## TODO: This potential has locking issues
   validates_with UniqueForOwnerValidator
 
-  can_by_role :destroy, :update, :update_links, :destroy_links,
-              roles: [ :owner, :collaborator ]
+  can_by_role :destroy, :update, :destroy_links, roles: [ :owner, :collaborator ]
+  can_by_role :update_links, roles: [ :owner, :collaborator, :contributor ]
   can_by_role :index, :show, public: true, roles: [ :owner, :collaborator, :viewer ]
```

Comment:
I assume we want to have the :contributor to be able to show / index the collections too? Or just rely on the collection being made public to provide view rights?